### PR TITLE
Fixes for QMP communicators

### PR DESCRIPTION
**Pull request by mathiaswagner**
Mostly fixes issues observed when running Chroma after #1020.

### CODE REVIEW

**Comment by weinbe2** [comment]
Changes are straightforward and look good.

Some things I'm confused about @hummingtree, and should've asked about in your initial PR:

1. Why does QMP's `comm_rank_global` -> `QMP_get_node_number` work to always pick the _global_ rank zero, as opposed to `comm_rank` -> `QMP_comm_get_node_number`? I'd assume `QMP_get_node_number` would grab the current communicator as opposed to the default. Have you looked under the hood to make sure it'll always behave as expected? (And/or should I?)

2. I think there's a potential risk in the MPI backend: if `user_set_comm_handle` in the constructor is set to `true`, the "default" MPI world won't be `MPI_COMM_WORLD`, which runs afoul of the MPI backend's `comm_rank_global`baking in `MPI_COMM_WORLD`. A simple solution could be caching the default MPI world on the MPI backend `Communicator` constructor, similar to how we're saving some default state with the QMP backend. (This may also be helpful for the QMP backend re:my concern above.)

**Comment by weinbe2** [comment]
On a different note, checking against NERSC small looks good---all the deviations from the sample (CPU) output look familiar.

File: include/communicator_quda.h [hunk 1]
```diff
@@ -680,6 +680,9 @@ struct Communicator {
 
 #if defined(QMP_COMMS)
   QMP_comm_t QMP_COMM_HANDLE;
+  // A bool indicating if the QMP handle here is the default one, which we should not free at the end,
```
Comment (maddyscientist) [inline include/communicator_quda.h]: Comment here should be in doxygen markup

**Review by maddyscientist — APPROVED** [review]
Aside from my trivial comment on comment, this looks good.  I've done some testing with split-grid solves and also tested that `QUDA_RANK_VERBOSITY` works as expected here, and it does.  Signing off on this.

**Pull request merged**